MonoGame.Framework/Platform/Native/Platform.Interop.cs

```diff
@@ -0,0 +1,299 @@
+// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Input;
+using System.Runtime.InteropServices;
+
+namespace MonoGame.Interop;
+
+internal class MGHandleAttribute : System.Attribute
+{
+}
+
+internal enum EventType : uint
+{
+    Quit,
+
+    WindowMoved,
+    WindowResized,
+    WindowGainedFocus,
+    WindowLostFocus,
+    WindowClose,
+
+    KeyDown,
+    KeyUp,
+    TextInput,
+
+    MouseMove,
+    MouseButtonDown,
+    MouseButtonUp,
+    MouseWheel,
+
+    ControllerAdded,
+    ControllerRemoved,
+    ControllerStateChange,
```

Comment:
@mrhelmut @harry-cpp I opted for a simple 3 event design which seems to work well.

Reply:
Another question... should i expand this to support the MG `Joystick` APIs thru the same events?

What makes things a joystick vs gamepad?

I REALLY want to unify this to keep the native API simple.

@mrhelmut @harry-cpp

Reply:
> What makes things a joystick vs gamepad?

In XNA terms, as far as I'm aware Joysticks were DirectInput devices and GamePads were XInput devices (https://learn.microsoft.com/en-us/windows/win32/xinput/xinput-and-directinput).

GamePads are typically more standardized with 2 thumbsticks, 10 buttons and 2 triggers. So a flight sim controller with 20 buttons and 8 analogue values for the stick and engine throttles would not have XInput support and wouldn't appear as a GamePad. The XNA GamePad Enums also wouldn't exist for all the extra values. Vibration is handled differently as well (SetVibration left/right instead of force feedback implementation).

Also with GamePads potentially being wireless, XInput added a connected status and handles controller assignment swapping (for example when the batteries go flat, turning on another controller can replace the original number assignment instead of adding a second controller).

In MonoGame DesktopGL terms, DirectInput joysticks that are actually gamepads can be automatically mapped by SDL as GamePads. That is why PS4 controllers (DirectInput) work on DesktopGL but not on WindowsDX.

The joystick implementations for MonoGame currently do not exist for WindowsDX and Android (maybe others) as well.

Reply:
I think that we can consider gamepads anything that goes through SDL's game controller API, and anything that isn't is a joystick.